plugins/esi_node_provisioning.py

```diff
@@ -0,0 +1,31 @@
+from rally_openstack.task.scenario import scenario
+from rally import consts
+from rally.task import atomic
+from rally.task import validation
+import subprocess
+import time
+import os
+
+@validation.add("required_platform", platform="openstack", users=True)
+@scenario.configure(name="ScenarioPlugin.esi_node_provisioning")
+
+class EsiNodeProvisioning(scenario.Scenario):
+    def esi_node_provisioning(self, image=None, network=None, candidate=None, sshpublickey=None):
+        """Scenario to test 'openstack esi node provisioning' command."""
+        command = f"metalsmith deploy --resource-class baremetal --image {image} --network {network} --candidate {candidate} --ssh-public-key {sshpublickey}"
+        if image:
+            command += f" --image {image}"
+        if candidate:
+            command += f" --candidate {candidate}"
+        if network:
+            command += f" --network {network}"
+        if sshpublickey:
+            command += f" --ssh-public-key {sshpublickey}"
+
+        start_time = time.time()
+        output = os.system(command)
+        end_time = time.time()
+        execution_time = end_time - start_time
+
+    def run(self, image=None, network=None, candidate=None, sshpublickey=None):
```

Comment:
Defaulting parameters like this should only really be done if the parameter is optional; none of these are. I might suggest:

`def run(self, image, network, candidate, ssh_public_key=None, resource_class='baremetal'):`

Reply:
Sure will update that.